LocalPayment/src/main/java/com/braintreepayments/api/localpayment/LocalPaymentApi.kt

```diff
@@ -0,0 +1,84 @@
+package com.braintreepayments.api.localpayment
+
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.localpayment.LocalPaymentNonce.Companion.fromJSON
+import org.json.JSONException
+import org.json.JSONObject
+
+internal class LocalPaymentApi(private val braintreeClient: BraintreeClient) {
+    fun createPaymentMethod(
+        request: LocalPaymentRequest,
+        callback: LocalPaymentInternalAuthRequestCallback
+    ) {
+        val returnUrl =
+            braintreeClient.getReturnUrlScheme() + "://" + LocalPaymentClient.LOCAL_PAYMENT_SUCCESS
+        val cancel =
+            braintreeClient.getReturnUrlScheme() + "://" + LocalPaymentClient.LOCAL_PAYMENT_CANCEL
+
+        val url = "/v1/local_payments/create"
+        braintreeClient.sendPOST(url = url, data = request.build(returnUrl, cancel),
+            responseCallback = { responseBody: String?, httpError: Exception? ->
+                if (responseBody != null) {
+                    try {
+                        val responseJson = JSONObject(responseBody)
+                        val redirectUrl = responseJson.getJSONObject("paymentResource")
+                            .getString("redirectUrl")
+                        val paymentToken = responseJson.getJSONObject("paymentResource")
+                            .getString("paymentToken")
+
+                        val transaction =
+                            LocalPaymentAuthRequestParams(request, redirectUrl, paymentToken)
+                        callback.onLocalPaymentInternalAuthResult(transaction, null)
+                    } catch (e: JSONException) {
+                        callback.onLocalPaymentInternalAuthResult(null, e)
+                    }
+                } else {
+                    callback.onLocalPaymentInternalAuthResult(null, httpError)
+                }
+            })
+    }
+
+    fun tokenize(
+        merchantAccountId: String?, responseString: String?, clientMetadataID: String?,
```

Comment:
I'm not sure if detekt will flag having multiple params on a single line. It may help with readability to have them all on separate lines.